src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs

```diff
@@ -167,26 +171,26 @@ private Task DiscoverPeersAsync()
         }
 
         /// <summary>
-        /// Add peers to the address manager from the network DNS's seed nodes.
+        /// Add peers to discover from the network DNS seed nodes.
         /// </summary>
         private void AddDNSSeedNodes(List<NetworkAddress> peers)
         {
-            peers.AddRange(this.network.DNSSeeds.SelectMany(seed =>
+            foreach (var seed in this.network.DNSSeeds)
             {
                 try
                 {
-                    return seed.GetAddressNodes();
+                    var seedAddresses = seed.GetAddressNodes().Select(ip => new NetworkAddress(ip, this.network.DefaultPort));
```

Comment:
Is `DefaultPort` correct? the previous code did not specify a port.

Reply:
Yes if you look at the diff you'll see we created network addresses from the IP and default port. There's a select statement:

```.Select(d => new NetworkAddress(d, this.network.DefaultPort)));```